frontend/src/pages/user/admin/manageUsers/index.tsx

```diff
@@ -1,10 +1,80 @@
 import { Col, Row } from 'react-bootstrap';
-import UserTableRow from '@components/UserTableRow';
-import datas from '@pages/user/admin/UserData.json';
+import { useMutation } from '@tanstack/react-query';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { CheckMutateStatus, RouteOnNotOK } from '@lib/Status';
 import Pagination from '@components/Pagination';
+import UserTableRow from '@components/UserTableRow';
 import UserTableHeader from '@components/UserTableHeader';
+import { useEffect, useState } from 'react';
+
+interface ICreditCard {
+  CVV: string;
+  name: string;
+  card_number: string;
+  expiry_date: string;
+}
+
+interface IUser {
+  address: string;
+  credit_card: [ICreditCard];
+  email: string;
+  enabled: true;
+  icon_url: string;
+  name: string;
+  role: string;
+  username: string;
+}
 
 const ManageUser = () => {
+  const navigate = useNavigate();
+  const [isMore, setIsMore] = useState(true);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const itemLimit = 10;
+
+  const {
+    data: fetchedData,
+    status,
+    mutate,
+  } = useMutation({
+    mutationKey: ['adminGetUser'],
+    mutationFn: async (isPage: boolean) => {
+      if (!isPage) {
+        const newSearchParams = new URLSearchParams({
+          offset: '0',
+          limit: (itemLimit + 1).toString(), // request one more to check if there is more
+        });
+        setSearchParams(newSearchParams, { replace: true });
+      }
+      const resp = await fetch('/api/admin/user?' + searchParams.toString(), {
+        method: 'GET',
+        headers: { accept: 'application/json' },
+      });
+      RouteOnNotOK(resp, navigate);
+      const response = await resp.json();
+      if (response.length === itemLimit + 1) {
+        response.pop();
+        setIsMore(true);
+      } else {
+        setIsMore(false);
+      }
+      return response as IUser[];
+    },
```

Comment:
`GET` requests should use `useQuery` instead of `useMutate`

Reply:
this is cus i combine ``request when first enter`` and ``request when switch page`` together. 
and it's done by giving it a parameter ``isPage``.
is using ``useMutate`` has some side effect here? ( i thought it just like ``useQuery`` with enabled=false and allow params.

Reply:
I don't know if there's a difference between `mutate` and `refetch` but I think it's a semantic issue as I think `mutate` is supposed to create side effects

Reply:
ok ill change it this night